File: JAVA/Insert and Search in a Trie.java
static void insert(TrieNode root, String key) 
{
    for(char ch: key.toCharArray()){
        if(root.children[ch-'a'] == null){
            root.children[ch-'a'] = new TrieNode();
        }
        root = root.children[ch-'a'];
    }
    root.isEndOfWord = true;
}
static boolean search(TrieNode root, String key)
{
    for(char ch: key.toCharArray()){
        if(root.children[ch-'a'] == null){
            return false;
        }
        root = root.children[ch-'a'];
    }
    return root.isEndOfWord;
}

File: C++/Insert and Search in a Trie.cpp
class Solution
{
    public:
        void insert(struct TrieNode *root, string key)
        {
             if(key.length()==0){
        root->isLeaf=true;
        return;
    }
    int index=key[0]-'a';
    struct TrieNode*child;
    if(root->children[index]!=NULL){
        child=root->children[index];
    }else{
        child=getNode();
        root->children[index]=child;
    }
    insert(child,key.substr(1));
        }
        bool search(struct TrieNode *root, string key) 
        {
    if(root==NULL){
        return false;
    }
    if(key.length()==0){
        return root->isLeaf;
    }
    int index=key[0]-'a';
    struct TrieNode*child=root->children[index];
    return search(child,key.substr(1));
        }
};
